Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pickable.py

```diff
@@ -20,22 +20,64 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+import itertools
 import json
 import sys
-from re import search
+import re
 
 from .command import Command
 from .find import Info
 from .trace import Trace, Relationship, CommitsStory
 from datetime import datetime
-from webkitcorepy import arguments
+from webkitcorepy import arguments, run, string_utils
 from webkitscmpy import Commit, local
 
 
+def fuzzy_filter(string):
+    return re.compile(string)
+
+
+if sys.version_info > (3, 6):
+    try:
+        from rapidfuzz import fuzz
+
+        def fuzzy_filter(string, ratio=95):
+            return lambda content: fuzz.partial_ratio(string, content) > ratio
+    except ModuleNotFoundError:
+        pass
+
+
 class Pickable(Command):
     name = 'pickable'
     help = 'List commits in a range which can be cherry-picked'
 
+    DEFAULT_PATTERNS = [
+        'Versioning',
+        'Gardening',
+        'Build-Fix',
+        'Apply-Patch',
+    ]
+    PATTERNS = {
+        'Cherry-pick': [re.compile(r'[Cc]herry[- ][Pp]ick')],
+        'Versioning': [re.compile(r'[Vv]ersioning')],
+        'Gardening': [
+            re.compile(r'\[?GARDENING\]?'),
+            re.compile(r'\[?[Gg]ardening\]?'),
+            re.compile(r'\[?REBASELINE\]?'),
+            re.compile(r'\[?[Rr]ebaseline\]?'),
+            fuzzy_filter(r'is a constant failure'),
+            fuzzy_filter(r'is an almost constant failure'),
+            fuzzy_filter('are constant failures'),
+            fuzzy_filter('tests consistently failing'),
```

Comment:
I'm concerned these will filter out non-gardening commits, like https://github.com/WebKit/WebKit/commit/debb098c0323383b53f8a07901c55ee4762fcb20.

AFAICT, it is normal for us to file a bug including the phrase "is a constant failure", and it's also normal for an engineer to use that bug title when committing their fix.

Reply:
This is  good point.

I think we should add an extra filter on these commits, but I also think that filter should be in a different PR, because it involves something a bit out of scope of the goals of this change. I'm inclined to only consider a change a gardening change if the change itself matches these conditions and also only modifies files in `LayoutTests`...this feels like a bit of a cheat, because it's not necessarily applicable outside of WebKit, and it requires an extra `git` operation for each implicated commit, but also, I think it would fix this problem for good.

How would you feel if we removed:
```
fuzzy_filter(r'is a constant failure'),
fuzzy_filter(r'is an almost constant failure'),
fuzzy_filter('are constant failures'),
fuzzy_filter('tests consistently failing'),
```
and revisited those in a follow-up PR?

Reply:
That makes sense to me.